### DISCUSSION TITLE
How to build and release only the modified files using CI/CD pipeline in VSTS?

### DISCUSSION THREAD

**Discussion by pervaizdattoo** [community/community#63695] [Actions]
### Select Topic Area

Question

### Body

I am currently in the process of architecting a CI/CD pipeline using VSTS. Whenever a developer commits his changes to the solution in his local visual studio and syncs the changes with remote repository, it triggers the build and release process automatically. How do I release only the files modified by the developer to Azure platform? Is there a feature in VSTS to handle delta deployment?

**Comment by naveedalirehmani** [comment]
When using VSTS for build/release, it doesn't allow for building or releasing just the modified files because doing so might not capture the complete intention of the project (like if only a config file was changed).

However, you can customize your build process to do this:

Keep a record of files you upload or publish.
Compare this record with new changes.
Release only the files that have changed.
For more instructions, check the mentioned post.
https://stackoverflow.com/questions/42910678/incremental-build-vsts-for-files-changed-only-wordpress

**Comment by queenofcorgis** [comment]
GitHub's [Acceptable Use Policies](https://docs.github.com/en/site-policy/acceptable-use-policies/github-acceptable-use-policies) prohibits coordinated or inauthentic activity like rapid questions and answers. As a result, we'll be unmarking the answer and locking this post. Please note any future violations may result in a temporary or indefinite block from the Community. Thanks for understanding.
